Examples/Tests/ElectrostaticSphere/analysis_electrostatic_sphere.py

```diff
@@ -0,0 +1,115 @@
+#! /usr/bin/env python
+
+# Copyright 2019-2020 Luca Fedeli, Maxence Thevenet, Remi Lehe
+#
+#
+# This file is part of WarpX.
+#
+# License: BSD-3-Clause-LBNL
+
+"""
+This script tests the expansion of a uniformly charged sphere of electrons.
+
+The input file inputs_3d/inputs is used: it defines a sphere of charge with
+given initial conditions. The sphere of charge starts at rest and will expand
+due to Coulomb interactions. The test will check if the sphere has expanded at
+the correct speed and that the electric field is accurately modeled against a
+known analytic solution. While the radius r(t) is not analytically known, its
+inverse t(r) can be solved for exactly.
+"""
+import numpy as np
+from scipy.optimize import fsolve
+import sys
+import yt
+yt.funcs.mylog.setLevel(0)
+
+# Constants and initial conditions
+xmin = ymin = zmin = -0.5  #Box dimensions
+xmax = ymax = zmax = 0.5
+eps_0 = 8.8541878128e-12  #Vacuum Permittivity in C/(V*m)
+m_e = 9.10938356e-31  #Electron mass in kg
+q_e = -1.60217662e-19  #Electron charge in C
+pi = np.pi  #Circular constant of the universe
+r_0 = 0.1  #Initial radius of sphere
+dt = 1e-6  #Timestep used for simulation
+t_max = 3e-5  #Final time of simulation
+q_tot = -1e-15  #Total charge of sphere in C
+
+# Define functions for exact forms of v(r), t(r), Er(r) with r as the radius of
+# the sphere. The sphere starts with initial radius r_0 and this radius expands
+# with zero initial velocity. Note: v(t) and r(t) are not known analytically but
+# v(r) and t(r) can be solved analytically.
+#
+# The solution r(t) solves the ODE: r''(t) = a/(r(t)**2) with initial conditions
+# r(0) = r_0, r'(0) = 0, and a = q_e*q_tot/(4*pi*eps_0*m_e)
+#
+# Note: E might be saved on a staggered time grid shifted by dt/2 (check this!)
+v_exact = lambda r: np.sqrt((q_e*q_tot)/(2*pi*m_e*eps_0)*(1/r_0-1/r))
+t_exact = lambda r: np.sqrt(r_0**3*2*pi*m_e*eps_0/(q_e*q_tot)) \
+    * (np.sqrt(r/r_0-1)*np.sqrt(r/r_0) \
+       + np.log(np.sqrt(r/r_0-1)+np.sqrt(r/r_0)))
+func = lambda rho: t_exact(rho)-t_max+dt/2  #Objective function to find r(t_max)
+r_end = fsolve(func,r_0)[0]  #Numerically solve for r(t_max)
+E_exact = lambda r: np.sign(r)*(q_tot/(4*pi*eps_0*r**2)*(abs(r)>=r_end) \
+    + q_tot*abs(r)/(4*pi*eps_0*r_end**3)*(abs(r)<r_end))
+
+# Open plotfile specified in command line (commented lines for debugging)
+# if __name__ == "__main__":
+#     filename = 'diags/plotfiles/plt00030'
+# else:
+#     filename = sys.argv[1]
+filename = sys.argv[1]
+ds = yt.load( filename )
+
+# Load data pertaining to fields
+data = ds.covering_grid(level=0, left_edge=ds.domain_left_edge,
+                        dims=ds.domain_dimensions)
+Ex = data['Ex'].to_ndarray()
+Ey = data['Ey'].to_ndarray()
+Ez = data['Ez'].to_ndarray()
+nx = Ex.shape[0];  ny = Ey.shape[1];  nz = Ez.shape[2]
```

Comment:
we can afford those ;)
```suggestion
nx = Ex.shape[0]
ny = Ey.shape[1]
nz = Ez.shape[2]
```

compact version:
```suggestion
nx, ny, nz = Ex.shape
```